astronomer/providers/amazon/aws/operators/redshift_cluster.py

```diff
@@ -15,6 +16,80 @@
     from airflow.utils.context import Context
 
 
+class RedshiftDeleteClusterOperatorAsync(RedshiftDeleteClusterOperator):
+    """
+    Delete an AWS Redshift Cluster if cluster status is in `available` state.
+
+    :param cluster_identifier: ID of the AWS Redshift Cluster
+    :param aws_conn_id: aws connection to use
+    :param skip_final_cluster_snapshot: determines cluster snapshot creation
+    :param final_cluster_snapshot_identifier: name of final cluster snapshot
+    :param cluster_status_fetch_interval_seconds: interval seconds to wait for fetching cluster status
+    """
+
+    def __init__(
+        self,
+        *,
+        skip_final_cluster_snapshot: bool = True,
+        final_cluster_snapshot_identifier: Optional[str] = None,
+        cluster_status_fetch_interval_seconds: int = 10,
+        aws_conn_id: str = "aws_default",
+        poll_interval: float = 5.0,
+        **kwargs: Any,
+    ):
+        self.skip_final_cluster_snapshot = skip_final_cluster_snapshot
+        self.final_cluster_snapshot_identifier = final_cluster_snapshot_identifier
+        self.cluster_status_fetch_interval_seconds = cluster_status_fetch_interval_seconds
+        self.aws_conn_id = aws_conn_id
+        self.poll_interval = poll_interval
+        super().__init__(**kwargs)
+
+    def execute(self, context: "Context") -> None:
+        """
+        Logic that the operator uses to correctly identify which trigger to
+        execute, and defer execution as expected.
+        """
+        redshift_hook = RedshiftHook(aws_conn_id=self.aws_conn_id)
+        cluster_state = redshift_hook.cluster_status(cluster_identifier=self.cluster_identifier)
+        if cluster_state == "available":
+            self.defer(
+                timeout=self.execution_timeout,
+                trigger=RedshiftClusterTrigger(
+                    task_id=self.task_id,
+                    polling_period_seconds=self.poll_interval,
+                    aws_conn_id=self.aws_conn_id,
+                    cluster_identifier=self.cluster_identifier,
+                    operation_type="delete_cluster",
+                    skip_final_cluster_snapshot=self.skip_final_cluster_snapshot,
+                    final_cluster_snapshot_identifier=self.final_cluster_snapshot_identifier,
+                    cluster_status_fetch_interval_seconds=self.cluster_status_fetch_interval_seconds,
+                ),
+                method_name="execute_complete",
+            )
+        else:
+            self.log.warning(
```

Comment:
just thinking either we should log a warning or raise an error here

Reply:
It may happen that the task failed after deleting the cluster and it is retired. At that time we would get cluster_not_found and we would not want to task to fail then. Made a change to check the status; logging a warning if cluster is not found and otherwise, raising an exception like you suggested.